### PR TITLE
[2211,2212,2213] Testing fixes (4-3-stable)

### PR DESCRIPTION
**Pull request by FifthPotato**
Replace all instances of assertEquals with assertEqual 
Do not install unittest-xml-reporting==1.14.0
Version-sniff to avoid upgrading pip on Ubuntu 24 and later

Needed for testing. Tests pass.

### CODE REVIEW

**Review by korydraughn — APPROVED** [review]
Looks good.

Let's wait for one more approval before pounding.

**Review by alanking — APPROVED** [review]
Seems good for the time being. Let's # it

**Comment by FifthPotato** [comment]
Octothorpe'd.

**Comment by alanking** [comment]
@FifthPotato - Please check boxes in linked issues and cherry-pick to main. Thanks

**Pull request merged**
